nautobot/circuits/tests/test_filters.py

```diff
@@ -91,161 +92,65 @@ class CircuitTestCase(FilterTestCases.FilterTestCase, FilterTestCases.TenancyFil
-
-    def test_provider_network(self):
-        provider_network = ProviderNetwork.objects.all()[:2]
-        params = {"provider_network_id": [provider_network[0].pk, provider_network[1].pk]}
-        self.assertEqual(self.filterset(params, self.queryset).qs.count(), 2)
-
-    def test_type(self):
-        circuit_type = CircuitType.objects.first()
-        params = {"type_id": [circuit_type.pk]}
-        self.assertEqual(self.filterset(params, self.queryset).qs.count(), 3)
-        params = {"type": [circuit_type.slug]}
-        self.assertEqual(self.filterset(params, self.queryset).qs.count(), 3)
-
-    def test_status(self):
-        statuses = list(Status.objects.get_for_model(Circuit)[:2])
-        params = {"status": [statuses[0].slug, statuses[1].slug]}
-        self.assertEqual(
-            self.filterset(params, self.queryset).qs.count(),
-            self.queryset.filter(status__slug__in=params["status"]).count(),
-        )
-
-    def test_region(self):
-        params = {"region_id": [self.regions[0].pk, self.regions[1].pk]}
-        cts = CircuitTermination.objects.filter(site__region__in=params["region_id"])
-        circuit_count = cts.values_list("circuit", flat=True).count()
-        self.assertEqual(self.filterset(params, self.queryset).qs.count(), circuit_count)
-        params = {"region": [self.regions[0].slug, self.regions[1].slug]}
-        cts = CircuitTermination.objects.filter(site__region__slug__in=params["region"])
-        circuit_count = cts.values_list("circuit", flat=True).count()
-        self.assertEqual(self.filterset(params, self.queryset).qs.count(), circuit_count)
-
-    def test_site(self):
-        params = {"site_id": [self.sites[0].pk, self.sites[1].pk]}
-        cts = CircuitTermination.objects.filter(site__in=params["site_id"])
-        circuit_count = cts.values_list("circuit", flat=True).count()
-        self.assertEqual(self.filterset(params, self.queryset).qs.count(), circuit_count)
-        params = {"site": [self.sites[0].slug, self.sites[1].slug]}
-        cts = CircuitTermination.objects.filter(site__slug__in=params["site"])
-        circuit_count = cts.values_list("circuit", flat=True).count()
-        self.assertEqual(self.filterset(params, self.queryset).qs.count(), circuit_count)
+        factory.CircuitTypeFactory.create_batch(10)
+        factory.ProviderFactory.create_batch(10)
+        factory.CircuitFactory.create_batch(20)
+        factory.CircuitTerminationFactory.create_batch(2, has_region=True, term_side="A")
+        factory.CircuitTerminationFactory.create_batch(2, has_region=True, term_side="Z")
+        factory.CircuitTerminationFactory.create_batch(2, has_site=False, term_side="A")
+        factory.CircuitTerminationFactory.create_batch(2, has_site=False, term_side="Z")
+
+    def test_filters_generic(self):
+        for test in self.generic_filter_tests:
+            filter_name = test[0]
+            field_name = test[1] if len(test) == 2 else test[0]
```

Comment:
Perhaps:

```suggestion
            field_name = test[-1]
```

Reply:
I knew it felt wrong typing that. Thanks!